inter/pos/validators.go

```diff
@@ -29,108 +36,104 @@ type (
 	GValidators map[idx.StakerID]GenesisValidator
 )
 
+func NewBuilder() ValidatorsBuilder {
+	return ValidatorsBuilder{}
+}
+
+// Set appends item to Validator object
+func (vv ValidatorsBuilder) Set(id idx.StakerID, stake Stake) {
+	// add
+	if stake == 0 {
+		delete(vv, id)
+	} else {
+		vv[id] = stake
+	}
+}
+
+func (vv ValidatorsBuilder) Build() *Validators {
+	return NewValidators(vv)
+}
+
 // NewValidators return new pointer of Validators object
-func NewValidators() *Validators {
-	return &Validators{
-		indexes: make(map[idx.StakerID]int),
-		list:    make([]Stake, 0, 200),
-		ids:     make([]idx.StakerID, 0, 200),
+func NewValidators(values ValidatorsBuilder) *Validators {
```

Comment:
Do we need to create different Validators from the same ValidatorsBuilder?
If not, can you please simplify it, e.g., removing ValidatorsBuilder?